tower-http/src/propagate_header.rs

```diff
@@ -0,0 +1,87 @@
+use futures_util::ready;
+use http::{header::HeaderName, HeaderValue, Request, Response};
+use pin_project::pin_project;
+use std::future::Future;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower_layer::Layer;
+use tower_service::Service;
+
+#[derive(Clone, Debug)]
+pub struct PropagateHeaderLayer {
+    header: HeaderName,
+}
+
+impl PropagateHeaderLayer {
+    pub fn new(header: HeaderName) -> Self {
```

Comment:
would be nice to have at least a simple example for this...

Reply:
I've one now.

Before shipping anything I want to sure the docs are at least decent and that all middlewares have examples.